operator.go

```diff
@@ -0,0 +1,149 @@
+package cloudconfig
+
+import (
+	"bytes"
+	"html/template"
+	"strings"
+
+	"github.com/giantswarm/awstpr"
+)
+
+var (
+	awsFilesMeta []FileMetadata = []FileMetadata{
+		FileMetadata{
+			AssetPath:   "templates/operator/aws/decrypt-tls-assets",
+			Path:        "/opt/bin/decrypt-tls-assets",
+			Owner:       "root:root",
+			Permissions: 0700,
+		},
+	}
+	awsUnitsMeta []UnitMetadata = []UnitMetadata{
+		UnitMetadata{
+			AssetPath: "templates/operator/aws/decrypt-tls-assets.service",
+			Name:      "decrypt-tls-certs.service",
+			Enable:    true,
+			Command:   "start",
+		},
+	}
+	kvmUnitsMeta []UnitMetadata = []UnitMetadata{
+		UnitMetadata{
+			AssetPath: "templates/operator/kvm/add-route-bridge.service",
+			Name:      "add-route-bridge.service",
+			Enable:    true,
+			Command:   "start",
+		},
+	}
+)
+
+type FileMetadata struct {
+	AssetPath   string
+	Path        string
+	Owner       string
+	Permissions int
+}
+
+type FileAsset struct {
+	Metadata FileMetadata
+	Content  []string
+}
+
+type UnitMetadata struct {
+	AssetPath string
+	Name      string
+	Enable    bool
+	Command   string
+}
+
+type UnitAsset struct {
+	Metadata UnitMetadata
+	Content  []string
+}
+
+type OperatorExtension interface {
+	GetFiles() ([]FileAsset, error)
+	GetUnits() ([]UnitAsset, error)
+}
```

Comment:
Sure, see the comments I already wrote. 

- https://github.com/giantswarm/k8scloudconfig/pull/1#discussion_r102953453
- https://github.com/giantswarm/k8scloudconfig/pull/1#discussion_r102967916
- https://github.com/giantswarm/k8scloudconfig/pull/1#discussion_r103713751
- https://github.com/giantswarm/aws-operator/pull/14#discussion_r102486768

Ones again. Every code that is concerned about AWS specific logic, belongs to the AWS operator. So please put such code into this repository: https://github.com/giantswarm/aws-operator. Every code that is concerned about KVM specific logic, belongs to the KVM operator. So please put such code into this repository: https://github.com/giantswarm/kvm-operator. Right now I cannot give you a concrete file path, because this depends on the actual code. Anyway, this repository here is concerned with cloud config management. It should aim to provide an abstracted interface everybody can implement. So the implementation of the interface should be put into the separate operator repositories. That way this repository stays lightweight and abstracted. It does not care about AWS/KVM. It does only dumb an cares about cloud config generation. Every detail that is concerned about specialties of some operator should be provided as argument, where the argument represents an abstraction as well. You already defined the abstract interface `OperatorExtension`, which only cares about files and units. This is super awesome. So lets put the code that implements this interface into the operator that actually cares about the details of the files and units that it wants to see inside the cloud config. So this is my proposal. When you have another idea we can also talk about this.